micro-rdk/src/common/config_monitor.rs

```diff
@@ -1,19 +1,39 @@
 use super::app_client::{AppClient, AppClientError, PeriodicAppClientTask};
+#[cfg(feature = "provisioning")]
+use crate::common::{
+    grpc::ServerError,
+    provisioning::storage::{RobotConfigurationStorage, WifiCredentialStorage},
+};
 use crate::proto::app::v1::ConfigResponse;
 use futures_lite::Future;
+use std::fmt::Debug;
 use std::pin::Pin;
 use std::time::Duration;
 
-pub struct ConfigMonitor<'a> {
-    restart_hook: Option<Box<dyn FnOnce() + 'a>>,
+pub struct ConfigMonitor<'a, S>
+where
+    S: RobotConfigurationStorage + WifiCredentialStorage + Clone + 'static,
+    <S as RobotConfigurationStorage>::Error: Debug,
+    ServerError: From<<S as RobotConfigurationStorage>::Error>,
+    <S as WifiCredentialStorage>::Error: Sync + Send + 'static,
```

Comment:
This won't work unless `feature = "provisioning"` is enabled, but see #267 which offers a way out. Perhaps this change should wait on that one.

Reply:
@gvaradarajan Do I need to do anything to make provisioning work or will it just work after your PR?

Reply:
@allisonschiang - Please follow up on this comment once you have made adjustments to address the issue.